rpc.go

```diff
@@ -248,14 +248,23 @@ func GetCollectionInfo(r *hfile.Reader, keySampleSize int) (*gen.HFileInfo, erro
 		for next && found < keySampleSize {
 			if rand.Float64() < pr {
 				buf[found] = it.Key()
+				found++
 			}
 			next, err = it.Next()
 			if err != nil {
 				return nil, err
 			}
 		}
+		lastFound := found-1
```

Comment:
As discussed offline, I think my preference would be to just return a smaller `RandomKeys`, e.g. `buf[:found]`.